core/src/model/PlayerWeapon.java

```diff
@@ -0,0 +1,95 @@
+package model;
+
+import com.dongbat.jbump.*;
+
+import java.util.*;
+
+import static java.lang.Math.*;
+
+public abstract class PlayerWeapon {
+
+    private float weaponSpeed;
+    private float weaponDamage;
+    private float weaponRange;
+    private float weaponWidth;
+    private double weaponAngle;
+    private int weaponRotations;
+    private World<IEntity> world;
+    private CollisionFilter filter;
+    private Set<Entity> isKnown = new HashSet<Entity>();
+    private ArrayList<ItemInfo> items = new ArrayList<>();
+    private List<Entity> entities = new LinkedList<>();
+    private Set<Entity> seen = new HashSet<Entity>();
+
+    public PlayerWeapon(World<IEntity> world, float weaponDamage, float weaponRange, float weaponWidth, float weaponSpeed, float weaponAngle, int weaponRotations){
+        this.weaponDamage = weaponDamage;
+        this.weaponRange = weaponRange;
+        this.weaponWidth = weaponWidth;
+        this.weaponSpeed = weaponSpeed;
+        this.weaponAngle = weaponAngle;
+        this.weaponRotations = weaponRotations;
+        this.world = world;
+        filter = getFilter();
+    }
+
+    //in playercharachter the swords position shall update with the move command like player does
+    public void weaponSwing(int rotationStart, int rotationFinish, int animationpart, PlayerCharacter player){
```

Comment:
Passing player to method creates a circular dependence. Player should pass relevant values to the method instead of passing itself